server/controllers/cohortController.js

```diff
@@ -1,20 +1,50 @@
 import db from "../db";
 export const getAllCohorts = async (req, res) => {
-    try {
-      const q = "SELECT * FROM cohorts";
-      const result = await db.query(q);
-      const cohorts = result.rows;
-      res.status(200).json({ cohorts });
-    } catch (error) {
-      console.error("Error retrieving cohorts:", error);
-      res.status(500).send("Failed, Internal Server Error.");
-    }
-  };
+  try {
+    const q = "SELECT * FROM cohorts";
+    const result = await db.query(q);
+    const cohorts = result.rows;
+    res.status(200).json({ cohorts });
+  } catch (error) {
+    console.error("Error retrieving cohorts:", error);
+    res.status(500).send("Failed, Internal Server Error.");
+  }
+};
 
-export const getCohort = (req, res) => {
-    res.status(200).json("you can write your get function to get one cohort here");
+export const getCohort = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const q = "SELECT * FROM cohorts WHERE id = $1";
```

Comment:
i think here you need to return trainees regarding that cohort , in this way you will just get a row which shows the name of cohort which there is no puros for this , we want when click on cohort see the list of trainee for that cohort

Reply:
I send the response at line 22, this is the way when we get a cohort by Id, if we get that cohort so we can map the cohort (the list of trainees) in the frontend.

Reply:
just think about the query you write what it will back you ? with that how you can get back trainee?